backends/k8s/client.go

```diff
@@ -62,28 +65,36 @@ func NewK8sClient(kubeconfig string) (*Client, error) {
 		return nil, err
 	}
 
-	// Create the clientset
+	// Create the clientset (we use this for Node queries).
 	cs, err := kubernetes.NewForConfig(config)
 	if err != nil {
 		return nil, err
 	}
 	log.Debug(fmt.Sprintf("Created k8s clientSet: %+v", cs))
 
-	tprClient, err := buildTPRClient(config)
+	// Create the Calico backend client.  We use this to access all of the
+	// custom resources.
+	calicoK8sClient, err := calicok8s.NewKubeClient(&capi.KubeConfig{
+		Kubeconfig: kubeconfig,
+	})
+
 	if err != nil {
 		return nil, err
 	}
+
 	kubeClient := &Client{
-		clientSet: cs,
-		tprClient: tprClient,
+		clientSet:         cs,
+		calicoK8sClient:   calicoK8sClient,
+		nodeBgpPeerClient: resources.NewNodeBGPPeerClient(cs),
+		nodeBgpCfgClient:  resources.NewNodeBGPConfigClient(cs),
 	}
 
 	return kubeClient, nil
 }
 
 // GetValues takes the etcd like keys and route it to the appropriate k8s API endpoint.
 func (c *Client) GetValues(keys []string) (map[string]string, error) {
-	var kvps = make(map[string]string)
+	var vars = make(map[string]string)
```

Comment:
I believe this is called on a per-template basis.  This is mostly ok since most config only appears in a single template at a time.  But I think it does result in more queries than is necessary.

One potential way to fix this would be to query everything here and populate a kv map that we store.  We could then time-out the map after the interval period so that the next query would then do a re-population.